### ISSUE TITLE
Support resource arrays in usage file

### ISSUE THREAD

**Issue by aliscott** [infracost/infracost#528]
Terraform resources can be created using `for_each` or `count`. This creates an array of resources. We could support setting usage values for all elements of the array by using a wildcard like this:

```yaml
resource_usage:
  aws_cloudwatch_log_group.loggroup[*]:
    ...
```

I think expected functionality would be that the usage value should apply individually to each element of the array. But an alternative is that the usage value is the aggregated using across all elements of the array.

Relevant code: https://github.com/infracost/infracost/blob/master/internal/providers/terraform/parser.go#L101-L105

Unit tests can be added to cover the new functionality.
Also add a new integration test to cover this. This can be added in a file called `parser_test.go` (separate from parser_internal_test.go) within a package called `parser_test`.

**Comment by ekifel** [comment]
@alikhajeh1 hey, I want to work on it

**Comment by alikhajeh1** [comment]
Released today in v0.8.4 - thanks @ekifel!
